File: src/main/java/me/anton/sickcore/api/database/MongoConnection.java
package me.anton.sickcore.api.database;

import com.mongodb.ConnectionString;
import com.mongodb.MongoClientSettings;
import com.mongodb.client.MongoClient;
import com.mongodb.client.MongoClients;
import com.mongodb.client.MongoDatabase;
import lombok.Getter;
import me.anton.sickcore.api.utils.common.FileUtils;
import org.bson.Document;

import java.util.logging.Logger;

@Getter
public final class MongoConnection {

    private final MongoDatabase database;

    public MongoConnection() {
        Logger mongoLogger = Logger.getLogger("org.mongodb.driver");
        mongoLogger.setLevel(java.util.logging.Level.SEVERE);
        String uri = "mongodb://" + getOutput("username") + ":" + getOutput("password") + "@" + getOutput("address") + ":" + getOutput("port") + "/?authSource=" + getOutput("database");
        ConnectionString connectionString = new ConnectionString(uri);
        MongoClientSettings settings = MongoClientSettings.builder()
                .applyConnectionString(connectionString)
                .build();
        MongoClient mongoClient = MongoClients.create(settings);
        this.database = mongoClient.getDatabase((String) getOutput("database"));
    }

    private Object getOutput(String key){
        Document object = FileUtils.getAsDocument("mongo");
        return object.get(key);
    }

}

File: src/main/java/me/anton/sickcore/utils/mongo/MongoConnection.kt
package me.anton.sickcore.utils.mongo

import com.mongodb.ConnectionString
import com.mongodb.MongoClientSettings
import me.anton.sickcore.utils.FileUtils
import org.litote.kmongo.coroutine.CoroutineClient
import org.litote.kmongo.coroutine.coroutine
import org.litote.kmongo.reactivestreams.KMongo

class MongoConnection {

    private val credentials = FileUtils.getFileAsDocument("mongo")
    private val connectionString = ConnectionString("mongodb://${credentials.getString("username")}:${credentials.getString("password")}@${credentials.getString("address")}:${credentials.getInteger("port")}/?authSource=${credentials.getString("databaseName")}")
    private val settings = MongoClientSettings.builder().applyConnectionString(connectionString).build()
    val client: CoroutineClient = KMongo.createClient(settings).coroutine

    val db = client.getDatabase(credentials.getString("databaseName"))

}